src/modules/order-history/components/UserOrders.ts

```diff
@@ -0,0 +1,50 @@
+import { isServer, bottomVisible } from '@vue-storefront/core/helpers'
+import MyOrder from '@vue-storefront/core/compatibility/components/blocks/MyAccount/MyOrder'
+
+export default {
+  name: 'UserOrders',
+  mixins: [MyOrder],
+  data () {
+    return {
+      pagination: {
+        perPage: 10,
+        current: 0,
+        enabled: false
+      },
+      bottom: false,
+      lazyLoadOrdersOnscroll: true
+    }
+  },
+  watch: {
+    bottom (bottom) {
+      if (bottom) {
+        ++this.pagination.current
+      }
+    }
+  },
+  beforeMount () {
+    if (!isServer && this.lazyLoadOrdersOnscroll) {
+      window.addEventListener('scroll', () => {
+        this.bottom = bottomVisible()
+      }, {passive: true})
+    }
+  },
+  computed: {
+    ordersHistory () {
+      let items = []
+      items = this.$store.state.user.orders_history.items
```

Comment:
this should use vuex getter and not mutate it - just call commit. Modifying state that way is side-effect and we should avoid that